src/Widgets/LocationPopover.vala

```diff
@@ -0,0 +1,245 @@
+/*
+* Copyright 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+public class Tasks.LocationPopover : Tasks.EntryPopover<Tasks.Location?> {
+    private GtkChamplain.Embed map_embed;
+    private Gtk.SearchEntry search_entry;
+    private GLib.Cancellable search_cancellable;
+    private Granite.Widgets.ModeButton location_mode;
+    private Marker point;
+
+
+    public LocationPopover () {
+        Object (
+            icon_name: "location-inactive-symbolic",
```

Comment:
For this we probably want `mark-location-symbolic` since the location-active/inactive icons are meant to indicate the active status of location services and not that this object is a location
```suggestion
            icon_name: "mark-location-symbolic",
```